### ISSUE TITLE
TextBuffer.setTextInRange `undo` option is deprecated.

### ISSUE THREAD

**Issue by brycefranzen** [sompylasar/atom-cursor-indent#3]
The `undo` option is deprecated. Call groupLastChanges() on the TextBuffer afterward instead.
```
TextBuffer.setTextInRange (<embedded>:33539:22)
TextEditor.setTextInBufferRange (<embedded>:3552:35)
setIndentationForBufferRowWithoutUndo (/Users/brycefranzen/.atom/packages/atom-cursor-indent/lib/atom-cursor-indent.coffee:131:14)
autoIndentLine (/Users/brycefranzen/.atom/packages/atom-cursor-indent/lib/atom-cursor-indent.coffee:162:20)
runEditorSideEffectsForChangeCursorPosition (/Users/brycefranzen/.atom/packages/atom-cursor-indent/lib/atom-cursor-indent.coffee:287:9)
<unknown> (/Users/brycefranzen/.atom/packages/atom-cursor-indent/lib/atom-cursor-indent.coffee:30:14)
```

**Comment by sompylasar** [comment]
@brycefranzen Thank you for reporting. Which version of Atom are you using?

**Comment by brycefranzen** [comment]
Latest, 1.27
  
  
  

  
  
  
  
  
>

**Comment by sompylasar** [comment]
Thanks for waiting. I haven't yet upgraded, so cannot make and test the change quickly. Contributions are welcome! The package behavior is tricky, so please make sure to test it well.

**Comment by ErickXavier** [comment]
1.27.1 here!

**Comment by dowatugkins** [comment]
I just created PR 4 to fix this.

**Comment by sompylasar** [comment]
Thank you @dowatugkins, I approved the PR #4, I'll need to publish it.

**Comment by sompylasar** [comment]
Published as `v0.3.1`: https://github.com/sompylasar/atom-cursor-indent/releases/tag/v0.3.1

**Comment by sompylasar** [comment]
@dowatugkins FYI, it did break, so now I'll have to revert `0.3.1` changes and publish `0.3.2`.

### 0.3.0

Keeps indent while navigating empty lines. Undo is one-step.

![atom-cursor-indent@0.3.0](https://user-images.githubusercontent.com/498274/43852890-5a757e52-9af3-11e8-866b-839c4e76d1d1.gif)

### 0.3.1

Jumps back and forward while navigating empty lines. Undo is multi-step.

![atom-cursor-indent@0.3.1](https://user-images.githubusercontent.com/498274/43852966-897c9fdc-9af3-11e8-9ed8-c5d30f1b6b98.gif)

**Comment by sompylasar** [comment]
Published the revert as `v0.3.2`: https://github.com/sompylasar/atom-cursor-indent/releases/tag/v0.3.2

Reopening as the issue needs a different solution that does not break the plugin behavior.

**Comment by dowatugkins** [comment]
Figures it would need more work than their documentation said it would.

**Comment by dowatugkins** [comment]
#5 Fixes the particular issue you pointed out. I found a better example of how someone else solved the problem so I think this will fix it, but definitely test this further before merging. Let me know if it still doesn't work.

**Comment by sompylasar** [comment]
Published as `v0.3.3`: https://github.com/sompylasar/atom-cursor-indent/releases/tag/v0.3.3
